packages/elements-dev-portal/src/components/NodeContent/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { Docs } from '@stoplight/elements/components/Docs';
+import { MarkdownComponentsProvider } from '@stoplight/elements/components/MarkdownViewer/CustomComponents/Provider';
+import { CustomLinkComponent } from '@stoplight/elements/components/MosaicTableOfContents/types';
+import { PersistenceContextProvider } from '@stoplight/elements/context/Persistence';
+import { Box } from '@stoplight/mosaic';
+import { dirname, resolve } from '@stoplight/path';
+import { NodeType } from '@stoplight/types';
+import * as React from 'react';
+
+import { Node } from '../../interfaces/node';
+
+export type NodeContentProps = {
+  node: Node;
+  Link: CustomLinkComponent;
+};
+
+export const NodeContent = ({ node, Link }: NodeContentProps) => {
+  return (
+    <PersistenceContextProvider>
+      <NodeLinkContext.Provider value={[node, Link]}>
+        <MarkdownComponentsProvider value={{ link: LinkComponent }}>
+          <Box style={{ maxWidth: ['model'].includes(node.type) ? 1000 : undefined }}>
+            <Docs nodeType={node.type as NodeType} nodeData={node.data} />
+          </Box>
+        </MarkdownComponentsProvider>
+      </NodeLinkContext.Provider>
+    </PersistenceContextProvider>
+  );
+};
+
+const NodeLinkContext = React.createContext<[Node, CustomLinkComponent] | undefined>(undefined);
+
+const LinkComponent: React.FC<{ node: { url: string } }> = ({ children, node: { url } }) => {
+  const ctx = React.useContext(NodeLinkContext);
+
+  if (ctx) {
+    const [node, Link] = ctx;
+
+    const resolvedUri = resolve(dirname(node.uri), url);
+    const [resolvedUriWithoutAnchor, hash] = resolvedUri.split('#');
+    const edge = node.outbound_edges.find(edge => edge.uri === url || edge.uri === resolvedUriWithoutAnchor);
+
+    if (edge) {
+      return (
+        <Link to={edge.slug} hash={hash}>
+          {children}
+        </Link>
+      );
+    }
```

Comment:
What is this? 🤔

Reply:
@marcelltoth we can get the `http_service` from the `http_operation`'s inbound edges. Open to suggestions on how best to pass that down to the elements-core security badge

```ts
const httpServiceNode = httpOperationNode.inbound_edges.find(edge => edge.type === 'http_service');
```

Reply:
If we can easily get http service uri from operation node, and if we have access to that operation node (I assume we do through context) then we can do the same as we do now, just without the function that @philsturgeon mentioned, which could be removed making things easier.